h2o-extensions/xgboost/src/main/java/hex/tree/xgboost/XGBoostUtils.java

```diff
@@ -113,17 +114,33 @@ public static DMatrix convertFrameToDMatrix(Key<DataInfo> dataInfoKey,
             int cols = di.fullN();
             float[][] data = allocateDenseMatrix(nRows, di);
             long actualRows = denseChunk(data, chunks, f, vecs, w, di, cols, resp, weights, f.vec(response).new Reader());
+
+            StopWatch denseMatrixCopyingStopWatch = new StopWatch();
+            denseMatrixCopyingStopWatch.start();
+
             trainMat = new DMatrix(data, actualRows, cols, Float.NaN);
+
+            denseMatrixCopyingStopWatch.stop();
+            Log.info("Copying dense matrix with " + actualRows + " logical rows and "
+                    + cols + " logical columns "
+                    + "into native XGBoost finished in " + denseMatrixCopyingStopWatch.getTime() + " ms");
             assert trainMat.rowNum() == actualRows;
         }
 
+        StopWatch labelAndWeightsStopWatch = new StopWatch();
```

Comment:
Rest of the code base uses `water.util.Timer` class, I would stick to that unless there is a good reason.